openslides_backend/action/actions/group/update.py

```diff
@@ -12,5 +15,14 @@ class GroupUpdateAction(UpdateAction):
     """
 
     model = Group()
-    schema = DefaultSchema(Group()).get_update_schema(optional_properties=["name"])
+    schema = DefaultSchema(Group()).get_update_schema(
+        optional_properties=["name", "permissions"]
```

Comment:
Proposal: Use 2 fields for changing permissions instead of the database field permissions:
add_permissions and remove_permissions with a possible list of permissions
This way all updates can be made inside database transaction including the read and the user or developer can't accidentally remove permissions from the group.

Reply:
I think you misunderstood the goal of these changes. Currently, the client sends a `group.set_permission` for every click in the permission matrix to save the changes directly. This behaviour should be changed so that all changes are only saved after clicking a save button (see the linked issue). Of course, that would be possible with the current action, but would require cumbersome tracking of the changes in the client and calculating the payloads, which is unnecessary to implement and to execute each time. With your proposals, this would still be the case, just with another payload format.

> all updates can be made inside database transaction including the read 

I'm not sure I understand this part or what this has to do with the issue

> the user or developer can't accidentally remove permissions from the group

The whole point of the change is to give the user full control of the permissions. It is unnecessary to put articificial burdens in the workflow. And, if that's a real concern for you, we would need to change all other update actions as well: E.g. the update of a user's groups is done the same way (and almost all other list updates as well), where the client sends the full new list and the backend just validates it and writes it to the datastore.

Reply:
The [client issue](https://github.com/OpenSlides/openslides-client/issues/1925) explicitly demands tracking the changes and not send the whole matrix, that's done the best way with proposal 1.

> This way all updates can be made inside database transaction including the read and the user or developer can't accidentally remove permissions from the group.

Conflicts can be detected by transaction logic, if another user changed permissions of the group. Your way this will be silently overwritten.